genshi/compat.py

```diff
@@ -106,15 +126,13 @@ def build_code_chunk(code, filename, name, lineno):
                 co_name=name,
                 co_firstlineno=lineno,
             )
-        params =  [0, code.co_nlocals, code.co_kwonlyargcount,
-                  code.co_stacksize, code.co_flags | 0x0040,
-                  code.co_code, code.co_consts, code.co_names,
-                  code.co_varnames, filename, name, lineno,
-                  code.co_lnotab, (), ()]
-        if hasattr(code, "co_posonlyargcount"):
-            # PEP 570 added "positional only arguments"
-            params.insert(2, code.co_posonlyargcount)
-        return CodeType(*params)
+        # XXX: This isn't equivalent to the above "replace" code (we overwrite
+        # co_argcount, co_flags, co_freevars, and co_cellvars). Should one of
+        # them be changed?
```

Comment:
Thoughts?

Reply:
`build_code_chunk` is only called in `genshi/template/eval.py` with the following note:
```
    try:
        # We'd like to just set co_firstlineno, but it's readonly. So we need
        # to clone the code object while adjusting the line number
        return build_code_chunk(code, filename, name, lineno)
    except RuntimeError:
        return code
```
Based on the above note, I added the `.replace(...)` version of `build_code_chunk` fairly recently (well, post 3.8) and I think one day maybe the older version can just be removed and we can just call `code.replace` directly.